keps/sig-network/2112-netpol-namespace-name-selector/README.md

```diff
@@ -0,0 +1,371 @@
+with generally positive feedback on its usage.
+- Feature Gate is enabled by Default.
+
+#### GA Graduation
+
+- At least **four** NetworkPolicy providers (or CNI providers) support the The 
+name selector field.
+- The name selector has been enabled by default for at least 1 minor release.
+
+### Upgrade / Downgrade Strategy
+
+If upgraded no impact should happen as this is a new matching option and not 
+colliding with old ones.  There will potentially be some golang magic required 
+to convert objects at the type level to be flexible enough to support 
+different inputs, but this will use the K8s API Translation layer.
+
+If downgraded the CNI wont be able to look into the new field, as this does 
+not exists and network policies using this field will stop working.
+
+### Version Skew Strategy
+
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: NetworkPolicyNamespaceAsName
+    - Components depending on the feature gate: Kubernetes API Server
+  
+* **Does enabling the feature change any default behavior?**
+  No
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, but CNIs relying on the new field wont recognize it anymore
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  Nothing. Just need to check if the data is persisted in ``etcd`` after the 
+  feature is disabled and reenabled or if the data is missed
+
+* **Are there any tests for feature enablement/disablement?**
+ 
+ TBD
+ 
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  By looking at the kubernetes networkpolicys in the cluster
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+
+   CNI Provider metrics can be usedto confirm that creation of a new policy 
+   targeting a namespace name is working
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  Policy latency is currently not measured because its implemented by CNI providers 
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+ No
+
+## Implementation History
+
+None
+
+## Drawbacks
+
+We dont want to have > 1 way to select namespaces.
+
+## Alternatives
+
+A network policy operator could be created which translated a CRD into many networkpolicys on the fly, by watching namespaces and updating labels dynamically.  This would be a privileged container in a cluster and likely would not gain much adoption.
```

Comment:
I'm not at all sure what you're suggesting here

Reply:
added a small example and clarified it....  its that weird policy meta operator idea i keep shoping around :).